src/config.c

```diff
@@ -2935,11 +2943,77 @@ static int setConfigBindOption(standardConfig *config, sds *argv, int argc, cons
     if (argc == 1 && sdslen(argv[0]) == 0) argc = 0;
 
     /* Free old bind addresses */
-    for (j = 0; j < server.bindaddr_count; j++) {
-        zfree(server.bindaddr[j]);
+    for (j = 0; j < orig_bindaddr_count; j++) {
+        zfree(bindaddr[j]);
+    }
```

Comment:
not that important, just one is three lines and one is one line.
```suggestion
    for (j = 0; j < orig_bindaddr_count; j++) zfree(bindaddr[j]);
```